python/dolfinx/wrappers/fem.cpp

```diff
@@ -356,7 +357,9 @@ void declare_objects(nb::module_& m, std::string type)
              std::shared_ptr<const dolfinx::fem::FunctionSpace<U>> V)
           {
             new (bc) dolfinx::fem::DirichletBC<T, U>(
-                g, std::vector(dofs.data(), dofs.data() + dofs.size()), V);
+                std::move(g),
+                std::vector(dofs.data(), dofs.data() + dofs.size()),
+                std::move(V));
```

Comment:
`std::move` on `V` not required and will have no effect.

Reply:
`std::move` makes `V` an rvalue, but the `DirichletBC` constructor doesn't accept rvalues for the `FunctionSpace` argument (it takes it by value), which is why it may have no effect unless the compiler can do some copy elision.

Even if it did move the shared pointer, the cost of copying a shared pointer outside a hot loop is trivial. We shouldn't be making code syntactically more complex if there is no measurable performance benefit. Happy to reconsider if a realistic test can demonstrate a meaningful performance difference.